### PR TITLE
DM-2432 Removing sensor-library chapter. Ranamed device-management-library.

### PR DESCRIPTION
**Pull request by MaciekSoftwareAG**

### CODE REVIEW

File: content/reference/device-integrator-library-bundle/relay.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ In a managed object, a relay control model includes the state of the control. Wh
 }
 ```
 
+For devices with `c8y_Relay` fragment [relay control widget](/users-guide/cockpit#relay-control) can be used.
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:18]:
```suggestion
For devices with the `c8y_Relay` fragment the [Relay control widget](/users-guide/cockpit#relay-control) can be used.
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 2]
```diff
@@ -29,7 +31,18 @@ The operation representation is the same as the inventory representation:
 }
 ```
 
-### c8y\_RelayArray
+**SmartREST example**
+
+{{< product-c8y-iot >}} provides the 518 static response template:
+
+1. Device receives command via 518 static response template <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:38]:
```suggestion
1. The device receives the command via the 518 static response template<br>
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 3]
```diff
@@ -29,7 +31,18 @@ The operation representation is the same as the inventory representation:
...
+   `518,OPEN`
+2. Device sets operation status to EXECUTING <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:40]:
```suggestion
2. The device sets the operation status to EXECUTING<br>
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 4]
```diff
@@ -29,7 +31,18 @@ The operation representation is the same as the inventory representation:
...
+   `501,c8y_Relay`
+3. Device confirms successful execution by setting operation status to SUCCESSFUL <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:42]:
```suggestion
3. The device confirms successful execution by setting the operation status to SUCCESSFUL<br>
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 5]
```diff
@@ -43,6 +56,8 @@ In a managed object, the control model of an array of relays includes the state
 ]
 ```
 
+For devices with `c8y_Relay` fragment [relay array control widget](/users-guide/cockpit#relay-array-control) can be used.
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:59]:
```suggestion
For devices with the `c8y_Relay` fragment the [Relay array control widget](/users-guide/cockpit#relay-array-control) can be used.
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 6]
```diff
@@ -55,3 +70,14 @@ The operation representation is the same as the inventory representation:
   "OPEN"
 ]
 ```
+
+**SmartREST example**
+
+{{< product-c8y-iot >}} provides the 519 static response template:
+
+1. Device receives command via 518 static response template <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:78]:
```suggestion
1. The device receives the command via the 519 static response template<br>
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 7]
```diff
@@ -55,3 +70,14 @@ The operation representation is the same as the inventory representation:
...
+   `519,OPEN,CLOSED,CLOSED,OPEN`
+2. Device sets operation status to EXECUTING <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:80]:
```suggestion
2. The device sets the operation status to EXECUTING<br>
```

File: content/reference/device-integrator-library-bundle/relay.md [hunk 8]
```diff
@@ -55,3 +70,14 @@ The operation representation is the same as the inventory representation:
...
+   `501,c8y_RelayArray`
+3. Device confirms successful execution by setting operation status to SUCCESSFUL <br>
```
Comment (rssag) [inline content/reference/device-integrator-library-bundle/relay.md:82]:
```suggestion
3. The device confirms successful execution by setting the operation status to SUCCESSFUL<br>
```

File: content/reference/device-integrator-library-bundle/relay.md (hunk 4 above)
Comment (MaciekSoftwareAG) [reply content/reference/device-integrator-library-bundle/relay.md:42]: Probably I will never comprehend where to use "the" in a (the?) programming context.

**Review by rssag — APPROVED** [review]

**Comment by rssag** [comment]
what are the target versions?

**Comment by rssag** [comment]
@MaciekSoftwareAG 
this needs another approval by your team
what are the target versions?

File: content/reference/device-integrator-library-bundle/relay.md [hunk 9]
```diff
@@ -1,10 +1,10 @@
 ---
-weight: 40
-title: Common capabilities
+weight: 85
+title: Relay
 layout: redirect
 ---
```
Comment (l2p-) [inline content/reference/device-integrator-library-bundle/relay.md:5]: This whole file needs to be aligned with the rest of this documentation. This doc explains ho device integrations can implement certain functionalities into their devices. The focus should be on the fuctionality of monitoring and managing a single or multiple two-position switches on a device.
There are two separate relay relay widgets in Cockpit: Relay control (for a single switch) and Relay array control (for multiple switches) for each of these we need a chapter for how a device can announce the status to the platform and a section how controlling them works. All of those with JSON data model examples of the fragments a device should put into its MO and the operation, examples with SmartREST ad descriptions of steps expected from the device.

**Review by l2p- — CHANGES_REQUESTED** [review]

File: content/reference/device-integrator-library-bundle/relay.md (hunk 9 above)
Comment (l2p-) [reply content/reference/device-integrator-library-bundle/relay.md:5]: If you look at other segments in this doc they are structured in a very particular way that is specifically around explaining how to integrate certain functionalities into a device and what the platform expects from a device. A good example for this chapter is the network section. This is a much more complex operation but ultimately a very similar concept with announcing the status and the request chang operation which are always separate flows. Here is the structure that I would have expected:

### Relays
Text is ok example and link should be moved to the respective sub section

#### Single Relay
Text similar to: "To manage and monitor a single relay Cumulocity offers the Relay control widget"

##### Relay status
Text similar to: "Devices may announce their relay status by updating the c8y_Relay Fragment in their managed object"
<JSON example>
<SmartREST example>

##### Setting relay position
Text similar to: "Devices that support changing the relay position remotely may add the c8y_Relay operation to c8y_SupportedOperations. Then Users can request an update to the relay position in the Relay control widget. This creates an operation of fragment type c8y_Relay for the device.
<JSON example>
<Steps expected from the device>

1. set op to executing
2. Apply requested relay position
3. update relay status in MO
4. set op to successful

<SmartREST examples of all steps>

#### Multiple relays
<Repeat similar structure>

**Review by l2p- — CHANGES_REQUESTED** [review]
see comment

File: content/reference/device-integrator-library-bundle/relay.md (hunk 9 above)
Comment (MaciekSoftwareAG) [reply content/reference/device-integrator-library-bundle/relay.md:5]: Ok, tried one more time.

File: content/reference/device-integrator-library-bundle/relay.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+{{< product-c8y-iot >}} provides the 519 static response template:
+
+1. The device receives the command via the 519 static response template<br>
+   `519,OPEN,CLOSED,CLOSED,OPEN`
+2. The device sets the operation status to EXECUTING<br>
+   `501,c8y_RelayArray`
+3. The device sets its relays states.
+4. The device confirms successful execution by setting the operation status to SUCCESSFUL<br>
+   `503,c8y_RelayArray`
```
Comment (l2p-) [inline content/reference/device-integrator-library-bundle/relay.md:124]:
```suggestion
3. The device confirms successful execution by setting the operation status to SUCCESSFUL and sets the relay array state with the implicit parameters in the 503 static template<br>
   `503,c8y_RelayArray,OPEN,CLOSED,CLOSED,OPEN`
```

**Review by l2p- — CHANGES_REQUESTED** [review]
I have added one suggestion. Please also accept the suggestions from Richard

**Review by l2p- — APPROVED** [review]

**Pull request merged**